openff/toolkit/tests/test_molecule.py

```diff
@@ -2870,6 +2870,192 @@ def test_add_trivalent_lone_pair_virtual_site(self, molecule):
+        vsite_pos = molecule.compute_virtual_site_positions_from_atom_positions(
+            conformer
+        )
+        vsite_pos = vsite_pos / vsite_pos.unit
+
+        assert vsite_pos.shape == (1, 3)
+        assert np.allclose(vsite_pos, expected)
+
+    def test_monovalent_lone_pair_virtual_site_position(self):
+
+        import openff.toolkit.tests.test_forcefield
+
+        molecule = openff.toolkit.tests.test_forcefield.create_water()
+
+        # a 90 degree water molecule on the xy plane
+        molecule.add_conformer(
+            unit.Quantity(
+                np.array(
+                    [
+                        [1.0, 0.0, 0.0],
+                        [0.0, 0.0, 0.0],
+                        [0.0, 1.0, 0.0],
+                    ]
+                ),
+                unit.angstrom,
+            )
+        )
+
+        atom1 = molecule.atoms[0]
+        atom2 = molecule.atoms[1]
+        atom3 = molecule.atoms[2]
+        distance = 0.2 * unit.angstrom
+        out_of_plane_angle = 90 * unit.degree
+        in_plane_angle = 180 * unit.degree
+        vsite1_index = molecule.add_monovalent_lone_pair_virtual_site(
+            [atom1, atom2, atom3],
+            distance,
+            out_of_plane_angle=out_of_plane_angle,
+            in_plane_angle=in_plane_angle,
+            symmetric=False,
+        )
+        expected = np.array([[1.0, 0.0, -0.2]])
+
+        vsite_pos = molecule.compute_virtual_site_positions_from_conformer(0)
+        vsite_pos = vsite_pos / vsite_pos.unit
+        assert vsite_pos.shape == (1, 3)
+        assert np.allclose(vsite_pos, expected)
+
+        conformer = molecule.conformers[0]
+        vsite_pos = molecule.compute_virtual_site_positions_from_atom_positions(
+            conformer
+        )
+        vsite_pos = vsite_pos / vsite_pos.unit
+        assert vsite_pos.shape == (1, 3)
+        assert np.allclose(vsite_pos, expected)
+
+    def test_divalent_lone_pair_virtual_site_position(self):
+
+        import openff.toolkit.tests.test_forcefield
+
+        molecule = openff.toolkit.tests.test_forcefield.create_water()
+
+        # a 90 degree water molecule on the xy plane
+        molecule.add_conformer(
+            unit.Quantity(
+                np.array(
+                    [
+                        [1.0, 0.0, 0.0],
+                        [0.0, 0.0, 0.0],
+                        [0.0, 1.0, 0.0],
+                    ]
+                ),
+                unit.angstrom,
+            )
+        )
+
+        atom1 = molecule.atoms[0]
+        atom2 = molecule.atoms[1]
+        atom3 = molecule.atoms[2]
+        distance = 0.2 * unit.angstrom
+        out_of_plane_angle = 90 * unit.degree
+        vsite1_index = molecule.add_divalent_lone_pair_virtual_site(
+            [atom1, atom2, atom3],
+            distance,
+            out_of_plane_angle=out_of_plane_angle,
+            symmetric=False,
+        )
+        expected = np.array([[0.0, 0.0, -0.2]])
```

Comment:
All of the tests only have one orientation / particle. We should add/modify a test that expects multiple particles to ensure that the positions are returned correctly, and document whether a particular order is expected.